roachtest: typeorm failed

roachtest.typeorm [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/11645234?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/11645234?buildTab=artifacts#/typeorm) on master @ [ad4e53f6b4dda9a76c60d4fe52f6f9d7c6afde2a](https://github.com/cockroachdb/cockroach/commits/ad4e53f6b4dda9a76c60d4fe52f6f9d7c6afde2a):


```
(typeorm.go:160).func1: all attempts failed for building TypeORM: full command output in run_195141.721870164_n1_cd-mntdata1typeorm-n.log: COMMAND_PROBLEM: exit status 1
test artifacts and logs in: /artifacts/typeorm/run_1
```
<p>Parameters: <code>ROACHTEST_arch=amd64</code>
, <code>ROACHTEST_cloud=gce</code>
, <code>ROACHTEST_cpu=4</code>
, <code>ROACHTEST_encrypted=false</code>
, <code>ROACHTEST_ssd=0</code>
</p>
<details><summary>Help</summary>
<p>


See: [roachtest README](https://github.com/cockroachdb/cockroach/blob/master/pkg/cmd/roachtest/README.md)



See: [How To Investigate \(internal\)](https://cockroachlabs.atlassian.net/l/c/SSSBr8c7)



See: [Grafana](https://go.crdb.dev/p/roachfana/teamcity-11645234-1693979296-130-n1cpu4/1694027841361/1694029938747)

</p>
</details>
/cc @cockroachdb/sql-foundations
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*typeorm.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>


Jira issue: CRDB-31511

This test has been failing every day, on all branches, since the node version was bumped on Sep 1(#109896), where it would usually fail at around ~30 minutes (not posted in this issue)

Interestingly, a subsequent PR (#110065) resolved the error above, but now the test hangs at the below point, and then times out after 3 hours, thus still failing.

```
  7) database schema > generated columns > cockroachdb
       should remove data from 'typeorm_metadata' when table dropped:

      AssertionError: expected 4 to equal +0
      + expected - actual

      -4
      +0

      at /mnt/data1/typeorm/test/functional/database-schema/generated-columns/cockroachdb/generated-columns-cockroachdb.ts:248:50
      at processTicksAndRejections (node:internal/process/task_queues:95:5)
      at async Promise.all (index 0)

  8) database schema > generated columns > cockroachdb
       "after all" hook for "should remove data from 'typeorm_metadata' when table dropped":
     Error: Timeout of 90000ms exceeded. For async tests and hooks, ensure "done()" is called; if returning a Promise, ensure it resolves. (/mnt/data1/typeorm/build/compiled/test/functional/database-schema/generated-columns/cockroachdb/generated-columns-cockroachdb.js)
      at listOnTimeout (node:internal/timers:569:17)
      at processTimers (node:internal/timers:512:7)
```

@rafiss fyi - this is probably already on your radar

--- a/pkg/cmd/roachtest/tests/typeorm.go
+++ b/pkg/cmd/roachtest/tests/typeorm.go
@@ -25,7 +25,10 @@ import (
 )
 
 var typeORMReleaseTagRegex = regexp.MustCompile(`^(?P<major>\d+)\.(?P<minor>\d+)\.(?P<point>\d+)$`)
-var supportedTypeORMRelease = "0.3.17"
+
+// Use 0.3.18 from the upstream repo once it is released.
+const supportedTypeORMRelease = "remove-unsafe-crdb-setting"
+const typeORMRepo = "https://github.com/rafiss/typeorm.git"
 
 // This test runs TypeORM's full test suite against a single cockroach node.
 func registerTypeORM(r registry.Registry) {
@@ -117,7 +120,7 @@ func registerTypeORM(r registry.Registry) {
 			ctx,
 			t,
 			c,
-			"https://github.com/typeorm/typeorm.git",
+			typeORMRepo,
 			"/mnt/data1/typeorm",
 			supportedTypeORMRelease,
 			node,

